File: homework_2/prime.c
#include <stdio.h>
#include <stdbool.h>

bool prime(int initial) {
	bool isPrime = true;

	if(initial == 0 || initial == 1) {
		isPrime = false; //0 and 1 are not prime
	} else {	
	    for(int i = 2; i <= initial / 2; i++) {
			if(initial > 2 && initial % i == 0) {
				isPrime = false; //if the number is divisible by any number from 2 to its half then it isn't prime			
			}
		}
	}
	return isPrime;
}

int main() {
	int number;

	printf("Enter an integer: ");
	scanf("%d", &number);

	bool isPrime = prime(number);

	if(isPrime) {
		printf("%d is a prime number. \n", number);
	} else {
		printf("%d is NOT a prime number. \n", number);
	}

	return 0;
}

File: homework_2/prime.py
def isPrime(number):
    result = True
    i = 2
    
    if number == 0 or number == 1:
        result = False
    elif number == 2:
        return result
    else:
        while i < number / 2 + 1:
            if number % i == 0:
                result = False
                
            i += 1
        
    return result

def main():
    number = int(input("Enter a number: "))
    
    result = isPrime(number)
    
    if result == True:
        print("The number is prime.")
    else:
        print("The number is NOT prime.")
    
    return 0
    
if __name__ == "__main__":
    main()       
